astropy/visualization/tests/test_norm.py

```diff
@@ -281,6 +286,40 @@ def test_invalid_stretch(self):
             simple_norm(DATA2, stretch="invalid")
 
 
+@pytest.mark.skipif(not HAS_PLT, reason="requires matplotlib.pyplot")
+def test_simplenorm():
+    data = np.arange(25).reshape((5, 5))
+    snorm = SimpleNorm("linear", percent=99)
+    norm = snorm(data)
+    assert isinstance(norm, ImageNormalize)
+    assert_allclose(norm(data), simple_norm(data, "linear", percent=99)(data))
+
+
+@pytest.mark.skipif(not HAS_PLT, reason="requires matplotlib.pyplot")
+def test_simplenorm_imshow():
+    import matplotlib.pyplot as plt
+    from matplotlib.image import AxesImage
+
+    data = np.arange(25).reshape((5, 5))
+    fig, ax = plt.subplots()
+    snorm = SimpleNorm("sqrt", percent=99)
+    axim = snorm.imshow(data, ax=ax)
+    assert isinstance(axim, AxesImage)
+    keys = ("vmin", "vmax", "stretch", "clip", "invalid")
+    for key in keys:
+        assert getattr(axim.norm, key) == getattr(snorm(data), key)
+
+    plt.clf()
```

Comment:
`clf` is discouraged (though not formally deprecated)
https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.FigureBase.clf

```suggestion
    fig.clear()
```

Reply:
Fixed.  I also fixed this in the tests for the `imshow_norm` function.